core/services/pipeline/common_eth.go

```diff
@@ -90,12 +96,173 @@ func parseETHABIString(theABI []byte, isLog bool) (name string, args abi.Argumen
 		return "", nil, nil, errors.Errorf("bad ABI specification: %v", theABI)
 	}
 	name = string(bytes.TrimSpace(matches[0][1]))
-	if len(name) == 0 {
-		if isLog {
-			return "", nil, nil, errors.Errorf("bad ABI specification: no event name")
-		}
-		return "", nil, nil, errors.Errorf("bad ABI specification: no method name")
-	}
 	args, indexedArgs, err = parseETHABIArgsString(matches[0][2], isLog)
 	return name, args, indexedArgs, err
 }
+
+func convertToETHABIType(val interface{}, abiType abi.Type) (interface{}, error) {
```

Comment:
Is this intended to mimic the old behavior?

Reply:
It's intended to be a bit more expansive, particularly given that we're doing multiword responses now. It supports all sorts of exotic types, such as `[][69]bool` and `[420][]address` (those are Optimism chain IDs, don't get any ideas)

Reply:
Is it backwards compatible, do we need the script to take care of this if not?